test/utils/fresh.go

```diff
@@ -0,0 +1,50 @@
+package utils
+
+import (
+	"testing"
+)
+
+type Refresh struct {
+	TestRefreshServices    bool
+	TestRefreshConfigPaths bool
+}
+
+func TestRefresh(t *testing.T, snapName string, conf Refresh) {
+	t.Run("refreshing", func(t *testing.T) {
+		if conf.TestRefreshServices {
+			testRefreshServices(t, snapName)
+		}
+		if conf.TestRefreshConfigPaths {
+			testRefreshConfigPaths()
+		}
+	})
+}
+
+func testRefreshServices(t *testing.T, snapName string) {
+	const (
+		upgradedVersion = "latest/beta"
+		platformSnap    = "edgexfoundry"
+	)
+
+	t.Cleanup(func() {
+		if LocalSnap != "" {
+			SnapInstallFromFile(t, LocalSnap)
+		} else {
+			SnapInstallFromStore(t, platformSnap, ServiceChannel)
+		}
```

Comment:
I think this should remove and then install. Otherwise it will act like an upgrade which may fail.